cli/command/image/push.go

```diff
@@ -66,5 +68,9 @@ func RunPush(dockerCli command.Cli, opts pushOptions) error {
 	}
 
 	defer responseBody.Close()
-	return jsonmessage.DisplayJSONMessagesToStream(responseBody, dockerCli.Out(), nil)
+	if !opts.quiet {
+		return jsonmessage.DisplayJSONMessagesToStream(responseBody, dockerCli.Out(), nil)
```

Comment:
Looks like actual push errors are produced here, so skipping this caused errors to not be printed, and `docker push` _always returning a zero-status exit code ("success") when using `--quiet`

See https://github.com/docker/cli/issues/2284